t3.c

```diff
@@ -498,7 +498,8 @@ int main(int argc, char *argv[]) {
 
   FILE *logfile = fopen(logfile_name, "w");
   if (!logfile) {
-    perror("Error opening logfile");
+    fprintf(stderr, "Error opening logfile %s %s\n", logfile_name,
```

Comment:
Good idea!

Previously, the error message appeared like this:
```
% ./result-t3/bin/t3 -t /not/found -- sh -c "echo err 1>&2; echo out"
Error opening logfile: No such file or directory
```

So the only change I'd make would be to keep the formatting closer to the original and use single quotes surrounding the filename:
```suggestion
    fprintf(stderr, "Error opening logfile '%s': %s\n", logfile_name,
```

With this version the error looks like this:
```
% ./result-t3/bin/t3 -t /not/found -- sh -c "echo err 1>&2; echo out"
Error opening logfile '/not/found': No such file or directory
```